docs/guide/http_cache.md

```diff
@@ -72,573 +80,753 @@ fos_http_cache:
+this work.
 
-    Note that in `config/ezplatform.yaml`, the `purge_servers` setting is an array
-    while the `HTTPCACHE_PURGE_SERVER` environment variable should be a string.
+It also varies on `Authorization` to cover any possible basic auth headers in case that is used over sessions for some requests.
 
-    ##### Fastly service ID and API token
+!!! note "Problems with stale user hash"
 
-    You also need to provide your Fastly service ID and API token in the configuration.
+    If you notice issues with stale hash usage, before you disable this cache make sure login/logout always
+    generates new session IDs and performs a full redirect in order to make sure no requests are being made using stale
+    user context hashes.
 
-    The service ID can be obtained by logging in on http://fastly.com and clicking `CONFIGURE` in the top menu,
-    then `Show service ID` at the top left of the page.
+!!! warning "Known limitations of the user context hash"
 
-    See [this Fastly guide](https://docs.fastly.com/guides/account-management-and-security/using-api-tokens) for
-    instructions on how to generate a Fastly API token.
-    The token needs `purge_select` and `purge_all` scope.
+    If you are using URI-based SiteAccesses matching on a multi-repository installation (multiple databases). The default
+    SiteAccess on the domain needs to point to the same repository (database), because `/_fos_user_context_hash` is not
+    SiteAccess-aware by default (see `ezpublish.default_router.non_siteaccess_aware_routes` parameter). This in turn is
+    because reverse proxy does not have knowledge about SiteAccesses and it's not passing the whole URL in order to be
+    able to cache the user context hash response.
 
-    You may specify service ID and token:
+    Only known workaround would be to make it SiteAccess aware, and have custom VCL logic tied to your SiteAccess
+    matching with Varnish/Fastly, in order to send SiteAccess prefix as URI.
 
-    - using the `service_id` and `key` settings (sub elements of "fastly") in `config/packages/ezplatform.yaml`
-    - by setting the parameters `fastly_service_id` and `fastly_key` in `config/packages/ezplatform-http-cache-fastly.yaml`
-    - by setting the environment variables `FASTLY_SERVICE_ID` and `FASTLY_KEY`
+##### Default options for FOSHttpCacheBundle defined in eZ Platform
 
-    Unless you need different settings per SiteAccess it is recommended to either use `config/packages/ezplatform.yaml`
-    or the environment variables.
+The following configuration is defined in eZ Platform by default for FOSHttpCacheBundle.
+Typically, you should not override these settings unless you know what you are doing.
 
-    ##### Clear the cache
+``` yaml
+fos_http_cache:
+    proxy_client:
+        default: varnish
+        varnish:
+            http:
+                servers: ['$http_cache.purge_servers$']
```

Comment:
Q: does this dynamic configuration still work? AFAIK it's always gonna be the one from `default` scope at compile time and having it here might lead to incorrect expectation by an integrator. Not sure what would be an alternative in this case though, except static container parameter.

Reply:
Good point, it should be 

```suggestion
                servers: ['%purge_server%']
```

Reply:
_Ignore prev comment_

Good catch, then we have an issue in v3, what I have pasted here comes directly from: https://github.com/ezsystems/ezplatform-http-cache/blob/master/src/Resources/config/fos_http_cache.yml


_Side: Have someone grepped fro usage of `$` config in 3.x for EE / Commerce?_

Reply:
@alongosz Have this been reported already or How should we report this to engineering?

Reply:
> _Side: Have someone grepped fro usage of `$` config in 3.x for EE / Commerce?_

We've tested dynamic settings removal, so yes, but indeed it seems we've missed some configs.

> @alongosz Have this been reported already or How should we report this to engineering?

Recently (surprisingly all at once, though the issue exists since 3.0) there have been separate per-case reports of some things related to dynamic configuration not working.

We should indeed look at the bigger picture and grep the entire configuration. I need to sync with @lserwatka on that.

> 
> Also: Can we move forward with this PR despite this? After all the block here is just reflecting what's in the product.

Hmm, it depends how broken it is. If that settings injects default context (as I implied in my first comment), then it's fine, we'll probably catch it when grepping. But... the recent reported cases (for different configs) showed us that it's injected as-is, so in this case it would be set to `$http_cache.purge_servers$` verbatim. If that's the case here as well, we should provide something static as a temporary fix (or make an actual fix so the config works - probably better).

Reply:
Grouped as the Epic [EZP-31950](https://jira.ez.no/browse/EZP-31950). @mikadamczyk is looking into this now.

Reply:
I checked how this works. I set different servers for two SiteAccesses. The dynamic setting is parsed on each request in `ezplatform-http-cache/src/ProxyClient/HttpDispatcherFactory.php:39`

```
        foreach ($servers as $server) {
            if (!$this->dynamicSettingParser->isDynamicSetting($server)) {
                $allServers[] = $server;
                continue;
            }

            $settings = $this->dynamicSettingParser->parseDynamicSetting($server);
            $configuredServers = $this->configResolver->getParameter(
                $settings['param'],
                $settings['namespace'],
                $settings['scope']
            );
            $allServers = array_merge($allServers, (array)$configuredServers);
        }
```

For each SiteAccess I received a correct server URLs

Reply:
Thank you for checking @mikadamczyk, looks like in this case false alarm. We can resolve this.